code/services/SAMLConfiguration.php

```diff
@@ -101,29 +101,67 @@ public function asArray()
             // Indicates a requirement for the NameID received by
             // this SP to be encrypted.
             'wantNameIdEncrypted' => false,
-
             // Algorithm that the toolkit will use on signing process. Options:
             //  - 'http://www.w3.org/2000/09/xmldsig#rsa-sha1'
             //  - 'http://www.w3.org/2000/09/xmldsig#dsa-sha1'
             //  - 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha256'
             //  - 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha384'
             //  - 'http://www.w3.org/2001/04/xmldsig-more#rsa-sha512'
             'signatureAlgorithm' => $signatureAlgorithm,
-
             // Authentication context.
             // Set to false and no AuthContext will be sent in the AuthNRequest,
             // Set true or don't present thi parameter and you will get an AuthContext 'exact' 'urn:oasis:names:tc:SAML:2.0:ac:classes:PasswordProtectedTransport'
             // Set an array with the possible auth context values: array ('urn:oasis:names:tc:SAML:2.0:ac:classes:Password', 'urn:oasis:names:tc:SAML:2.0:ac:classes:X509'),
-            'requestedAuthnContext' => [
-                'urn:federation:authentication:windows',
-                'urn:oasis:names:tc:SAML:2.0:ac:classes:Password',
-                'urn:oasis:names:tc:SAML:2.0:ac:classes:X509',
-            ],
+            'requestedAuthnContext' => $this->getrequestedAuthnContext(),
             // Indicates if the SP will validate all received xmls.
             // (In order to validate the xml, 'strict' and 'wantXMLValidation' must be true).
             'wantXMLValidation' => true,
         ];
 
+        if (isset($security['requestedAuthnContextComparison'])) {
+            // Allows the authn comparison parameter to be set, defaults to 'exact' if
+            // the setting is not present.
+            // better | exact | maximum | minimum
+            $conf['security']['requestedAuthnContextComparison'] = $security['requestedAuthnContextComparison'];
+        }
+
         return $conf;
     }
+
+    private function getRequestedAuthnContext()
+    {
+        $security = $this->config()->get('Security');
```

Comment:
For config props we prefer lowerCamelCase (psr-2), rather than TitleCase.

I was looking for a `private static $Security = 'somedefault';` but I couldn't see any.